api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java

```diff
@@ -1,26 +1,129 @@
 package org.openmrs.module.fhir.api.diagnosticreport.handler;
 
+import ca.uhn.fhir.model.dstu2.composite.CodingDt;
+import ca.uhn.fhir.model.dstu2.resource.DiagnosticReport;
+import ca.uhn.fhir.model.dstu2.resource.Patient;
+import ca.uhn.fhir.model.dstu2.resource.Practitioner;
+import ca.uhn.fhir.model.primitive.IdDt;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterRole;
+import org.openmrs.Person;
+import org.openmrs.Provider;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.context.ServiceContext;
+import org.openmrs.module.fhir.api.PatientService;
+import org.openmrs.module.fhir.api.PractitionerService;
 import org.openmrs.module.fhir.api.diagnosticreport.DiagnosticReportHandler;
-import org.openmrs.module.fhir.api.diagnosticreport.DiagnosticReportTemplate;
+import org.openmrs.module.fhir.api.impl.PatientServiceImpl;
+import org.openmrs.module.fhir.api.util.FHIRPatientUtil;
+import org.openmrs.module.fhir.api.util.FHIRPractitionerUtil;
 
-import ca.uhn.fhir.model.dstu2.resource.DiagnosticReport;
+import java.util.List;
 
 public class LaboratoryHandler extends AbstractHandler implements DiagnosticReportHandler {
-	
+
+	public LaboratoryHandler() {
+		super();
+	}
+
 	@Override
-	public DiagnosticReport generateFHIRDiagnosticReport(DiagnosticReportTemplate omrsDiagnosticReport) {
-		return null;
+	public DiagnosticReport getFHIRDiagnosticReport(DiagnosticReport diagnosticReport) {
+		// Get Obs and set as `Name`
+		// Get Obs and set as `Status`
+		// Get EncounterDateTime and set as `Issued` date
+		// Get Encounter Patient and set as `Subject`
+		// Get Encounter Provider and set as `Performer`
+		// Get EncounterType and Set `ServiceCategory`
+		// Get valueDateTime in Obs and Set `Diagnosis[x]->DateTime`
+		// Get valueDateTime in Obs and Set `Diagnosis[x]->Period`
+
+		// ObsSet set as `Result`
+		// Binary Obs Handler
+		return diagnosticReport;
 	}
-	
+
 	@Override
-	public DiagnosticReportTemplate generateOpenMRSDiagnosticReport(DiagnosticReport fhirDiagnosticReport) {
-		return null;
+	public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticReport) {
+		log.info("Laboratory Handler : Save FHIR Diagnostic Report");
+		Encounter omrsDiagnosticReport = new Encounter();
+
+		//Set ID if available
+		if (diagnosticReport.getId() != null) {
+			omrsDiagnosticReport.setUuid(diagnosticReport.getId().getIdPart());
+		}
```

Comment:
Errr. you don't need to set a UUID. it gets set automatically when the encounter is stored in DB, you can just remove this.